lib/libvarnishapi/vxp_lexer.c

```diff
@@ -58,6 +58,32 @@ vxp_add_token(struct vxp *vxp, unsigned tok, const char *b, const char *e)
 	vxp->t = t;
 }
 
+static void
+vxp_add_token(struct vxp *vxp, unsigned tok, const char *b, const char *e)
+{
+
+	/* Ignore first or consecutive EOI tokens */
```

Comment:
Would it be cleaner to define a EOQ (end of query) token, and leave EOI to define the end of input?

Reply:
I will give this a try. I anticipated this one wouldn't pass under the radar ;)

Reply:
I think the folding and stuff that happens here breaks the lexer and parser separation. We probably should have a EOL/NL token (not EOQ as I suggested), and let the lexer at the tokens it finds as normal. The separation of things into separate expressions to be OR'ed together, and the skipping of empty queries resulting from blank lines, belongs in the parser IMO.

Reply:
Marking the conversation as resolved, we can take this to the second iteration.